lib_eio/process.mli

```diff
@@ -0,0 +1,51 @@
+type status = Exited of int | Signaled of int | Stopped of int
+
+val pp_status : Format.formatter -> status -> unit
+
+class virtual t : object
+    inherit Generic.t
+    method virtual pid : int
+    method virtual status : status Promise.t
+    method virtual stop : unit
+end
+
+val pid : #t -> int
+(** The process ID *)
+
+val status : #t -> status Promise.t
+(** Checks the status of the subprocess, this will block waiting for the subprocess
+    to terminate or be stopped by a signal. *)
+
+val stop : #t -> unit
+(** Stop a running subprocess *)
+
+class virtual mgr : object
+    inherit Generic.t
+    method virtual spawn : 
+        sw:Switch.t -> 
+        ?cwd:Fs.dir Path.t ->
+        stdin:Flow.source ->
+        stdout:Flow.sink -> 
+        stderr:Flow.sink ->
+        string ->
+        string list ->
+        t
+    
+    method virtual spawn_detached :
+        ?cwd:Fs.dir Path.t ->
+        stdin:Flow.source ->
+        stdout:Flow.sink -> 
+        stderr:Flow.sink ->
+        string ->
+        string list ->
+        t
+end
+(** A process manager capable of spawning new processes. *)
+
+val spawn : sw:Switch.t -> #mgr -> ?cwd:Fs.dir Path.t -> stdin:Flow.source -> stdout:Flow.sink -> stderr:Flow.sink -> string -> string list -> t
+(** [spawn ~sw t cmd] creates a new subprocess that is connected to the
```

Comment:
Can we call this `mgr`, since `t` refers to a "subprocess"?
```suggestion
(** [spawn ~sw mgr ?cwd ~stdin ~stdout ~stderr cmd args] creates a new subprocess that is connected to the
```

Reply:
Yep, much clearer :)) Fixed in https://github.com/ocaml-multicore/eio/pull/330/commits/b02b37b3a96429b1e5d920ebc1d67803e8a31892